src/components/DropDown/DropDown.js

```diff
@@ -1,163 +1,322 @@
-import ClickOutHandler from 'react-onclickout'
-import { theme } from '../../theme'
-import { springs, unselectable } from '../../utils/styles'
-import { PublicUrl } from '../../providers/PublicUrl'
-import DropDownItem from './DropDownItem'
-
-import arrow from './assets/arrow-down.svg'
-
-const NON_BREAKING_SPACE = '\xa0'
-
-const { contentBackground, contentBorder, textPrimary } = theme
-
-const StyledDropDown = styled.div`
-  position: relative;
-  z-index: ${({ opened }) => (opened ? '2' : '0')};
-  display: ${({ wide }) => (wide ? 'flex' : 'inline-flex')};
-  flex-direction: column;
-  color: ${textPrimary};
-  white-space: nowrap;
-  box-shadow: 0 4px 4px 0 rgba(0, 0, 0, 0.03);
-  ${unselectable()};
-  &:focus {
-    outline: 0;
-  }
-`
-
-const DropDownItems = styled(animated.div)`
-  position: absolute;
-  z-index: 2;
-  top: calc(100% - 1px);
-  padding: 8px 0;
-  color: ${textPrimary};
-  background: ${contentBackground};
-  border: 1px solid ${contentBorder};
-  box-shadow: 0 4px 4px 0 rgba(0, 0, 0, 0.06);
-  border-radius: 3px;
-  list-style: none;
-`
-
-const BlockingLayer = styled(animated.div)`
-  position: absolute;
-  z-index: 2;
-  top: 0;
-  left: 0;
-  right: 0;
-  bottom: 0;
-`
-
-const DropDownActiveItem = styled(PublicUrl.hocWrap(DropDownItem))`
-  padding-right: 40px;
-  background: ${contentBackground};
-  background-image: url(${PublicUrl.styledUrl(arrow)});
-  background-repeat: no-repeat;
-  background-position: calc(100% - 15px) 50%;
-  border: 1px solid ${contentBorder};
-  border-radius: 3px;
-  &:hover,
-  &:focus {
-    color: inherit;
-  }
-  &:active {
-    color: ${textPrimary};
-  }
-`
-
-class DropDown extends React.Component {
-  static propTypes = {
-    items: PropTypes.arrayOf(PropTypes.node),
-    wide: PropTypes.bool,
-    active: PropTypes.number,
-    onChange: PropTypes.func,
-  }
-  static defaultProps = {
-    items: [],
-    wide: false,
-    active: 0,
-    onChange: () => {},
+import { ButtonBase } from '../Button/ButtonBase'
+import Popover from '../Popover/Popover'
+import { IconDown } from '../../icons'
+import { GU, RADIUS, textStyle } from '../../style'
+import { useTheme } from '../../theme'
+import { warnOnce, unselectable } from '../../utils'
+import { useViewport } from '../../providers/Viewport/Viewport'
+
+function useDropDown({
+  buttonRef,
+  items,
+  label,
+  onChange,
+  placeholder,
+  selected,
+}) {
+  const [selectedLabel, setSelectedLabel] = useState(placeholder || label)
+  const [opened, setOpened] = useState(false)
+
+  const handleClose = useCallback(() => {
+    // if the popover is opened and the user clicks on the button
+    // this handler was being called before the click handler, so the
+    // click handler was re-opening the popover, by having this on the
+    // next tick things happen in order.
+    setTimeout(() => setOpened(false), 10)
+
+    if (buttonRef.current) {
+      buttonRef.current.focus()
+    }
+  }, [setOpened])
```

Comment:
```suggestion
  }, [])
```

`useState`'s update function doesn't need to be included in dependency lists.